### ISSUE TITLE
spacemacs/jump-to-definition does not query lsp backend for new Python buffers

### ISSUE THREAD

**Issue by swinkels** [syl20bnr/spacemacs#14729]
#### Description :octocat:
spacemacs/jump-to-definition does not query the lsp backend for new Python buffers. Section *Observed behaviour* describes  the scenario that goes wrong.

#### Reproduction guide :beetle:
- Start Emacs
- Open a Python file, which causes lsp to be started
- In this Python file, `spacemacs-jump-handlers` is the list
  `((lsp-ui-peek-find-definitions :async t) evil-goto-definition dumb-jump-go)`
- Open a second Python file
- In this second Python file, `spacemacs-jump-handlers` is the list
  `(evil-goto-definition dumb-jump-go)`

*Observed behaviour:* :eyes: :broken_heart:
- Start Emacs
- Open a Python file, which causes lsp to be started
- In this Python file, execute `, gg` on a symbol and point jumps to the
  definition of the symbol as provided by the lsp backend
- Open a second Python file
- In this second Python file, execute `, gg` on a symbol point jumps to the
  import of this symbol in the same file; the lsp backend was not consulted

*Expected behaviour:* :heart: :smile:
- Start Emacs
- Open a Python file, which automatically starts lsp
- In this Python file, execute `, gg` on a symbol and point jumps to the
  definition of the symbol as provided by the lsp backend
- Open a second Python file
- In this second Python file, execute `, gg` on a symbol and point jumps to the
  definition of the symbol as provided by the lsp backend

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 679040fbf)
- Graphic display: nile
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(yaml auto-completion c-c++
      (c-c++ :variables c-c++-backend 'lsp-ccls)
      emacs-lisp javascript git helm html lsp lua markdown org outshine
      (python :variables python-backend 'lsp python-formatter 'black python-format-on-save t python-test-runner 'pytest)
      (shell :variables shell-default-height 30 shell-default-position 'bottom)
      syntax-checking treemacs)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND DBUS GSETTINGS GLIB NOTIFY INOTIFY GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS PDUMPER GMP

**Comment by swinkels** [comment]
I did some debugging and I think the following is going on.

When `lsp-mode` initializes, it adds a hook to `lsp-after-open-hook` that adds the appropriate lsp "find definition" function to the Spacemacs jump handlers of `spacemacs-jump-handlers`. However, there is also a Python mode hook that executes later, `spacemacs//init-jump-handlers-python-mode`, that resets `spacemacs-jump-handlers` back to default. So when you open a new Python file and use `spacemacs/jump-to-definition`, it will not use lsp to find the definition.             
                                                                                            
There is a workaround. The aforementioned `spacemacs//init-jump-handlers-python-mode` does honor the jump handlers of `spacemacs-jump-handlers-python-mode`. As a workaround we add the lsp "find definition" function to that list:                                    
``` python                                                                                            
(add-to-list 'spacemacs-jump-handlers-python-mode '(lsp-ui-peek-find-definitions :async t)) 
```
and `spacemacs-jump-handlers` will contain the lsp "find definition" function.

**Comment by thanhvg** [comment]
Can you update to latest dev, it should be fixed with this f136f465abf599f6869b178e9338c72053444aec

**Comment by swinkels** [comment]
I've updated to the latest develop and I can confirm that this has been fixed.

Apologies for wasting your time. It's only a few weeks ago that I updated my local Spacemacs and I didn't realize that that version might not be up-to-date enough.
